AgenciaAutos/AgenciaAutos/Classes/Container.cs

```diff
@@ -0,0 +1,147 @@
+﻿
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace AgenciaAutos.Classes
+{
+    class Container<T> where T : IEntity
+    {
+        public List<T> list { get; set; }
+
+        public Container()
+        {
+            list = new();
+        }
+
+        public void Create(T item, string filePath)
+        {
+
+            if (item != null)
+            {
+                
+                ReadJson(filePath);
+                list.Add(item);
+                WriteFile(filePath);
+            }
+
+        }
+
+        public void WriteFile(string filePath)
+        {
+            var options = new JsonSerializerOptions { WriteIndented = true };
+
+            string json = JsonSerializer.Serialize(list, options);
+
+            using (StreamWriter sw = new StreamWriter(filePath))
+            {
+                sw.Write(json);
+                sw.Close();
+            }
+
+        }
+
+        public void ReadJson(string filePath)
+        {
+            list = new List<T>();
+            string json = "";
+
+            if (File.Exists(filePath))
+            {
+                json = ReadFile(filePath);
+            }
+
+            if (!string.IsNullOrEmpty(json))
+            {
+                List<Rental> entities = JsonSerializer.Deserialize<List<Rental>>(json);
+                
+            }
+
+        }
+
+        public string ReadFile(string filePath)
+        {
+            return File.ReadAllText(filePath);
+        }
+
+
+        public T Update(T item, string filePath)
+        {
+            T obj = default;
+            ReadJson(filePath);
+            List<T> newList= new List<T>();
+
+            list.ForEach(x =>
+            {
+                    if (item.Id == x.Id)
+                    {
+                        newList.Add(item);
+                        obj = item;
+                    }
+                    else
+                    {
+                        newList.Add(x);
+                    }
+            });
```

Comment:
technically this is not _wrong_ but I think a normal foreach loop would read a lot better.
also, let's see if I understand the logic correctly: you are creating a new list based on the existing list + the new item, right? what if the entity isn't found?
also, did you try just updating the properties of the entity when found?

Reply:
I create a new list with the elements of the old list but if the current element in the foreach  is that i wanna change i put the updated element in its place.